userapi/src/main/java/controller/RouterVerticle.java

```diff
@@ -0,0 +1,148 @@
+package controller;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.json.Json;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.TimeoutHandler;
+import model.D42Model;
+import model.IDataModel;
+import model.Profile;
+
+import java.util.Set;
+
+/**
+ * Routes requests to their respective handlers
+ * Created by rohit.patiyal on 18/01/17.
+ */
+public class RouterVerticle extends AbstractVerticle {
+
+    private static final long REQ_TIMEOUT = 2000;
+    private IDataModel dataModel = null;
```

Comment:
As suggest, I am refractoring it to ProfileDiscoveryAPI and ProfileDiscoveryAPIImpl for the interface and the implementation respectively.